source/common/http/http2/metadata_decoder.cc

```diff
@@ -85,6 +86,7 @@ bool MetadataDecoder::onMetadataFrameComplete(bool end_metadata) {
 }
 
 bool MetadataDecoder::decodeMetadataPayloadUsingNghttp2(bool end_metadata) {
+#ifdef ENVOY_NGHTTP2
   Buffer::RawSliceVector slices = payload_.getRawSlices();
```

Comment:
@birenroy I'm not sure how-if this works for oghttp2 if we actually remove nghttp2 from the build

I started off creating this build to slim down Envoy Mobile but I think it'd be helpful if you picked up a follow up where you made sure all the Envoy tests work with --define=nghttp2=disabled

Reply:
When `envoy.reloadable_features.http2_decode_metadata_with_quiche` is true, Envoy will decode metadata using QUICHE. Are you looking for something above and beyond that reloadable feature?

Reply:
nope that's what I missed.  I don't know why that didn't get a removal bug  - looks like it's over a year old.  I'll take care of it